source/deployment/permissions-backend.rst

```diff
@@ -0,0 +1,104 @@
+Advanced Permissions: Backend Infrastructure
+=============================================
+
+This document outlines the backend server infrastructure for permissions in Mattermost and is recommended only for technical Admins or developers looking to make modifications to their installation.
+
+
+.. note::
+
+  The contents of this document apply to Mattermost Server version 5.0 and later. 
+
+
+.. contents::
+  :backlinks: top
+  :local:
+  
+Entity Definitions
+--------------------
+
+Permissions
+~~~~~~~~~~~~
+
+A **permission** describes a permitted action which may be carried out on an object. It describes the action that users who have been granted the permission may perform in the context in which they have been granted it.
+
```

Comment:
`users who have been assigned roles granted the permission` `in the context in which they have been assigned the role granting the permission`.

It's a mouthful but otherwise it suggests that users can be directly granted permissions which is then immediately contradicted in the line below defining roles.